mewarpx/mewarpx/sim_control.py

```diff
@@ -93,3 +98,25 @@ def check_criteria(self):
             logger.info(terminate_statement)
 
         return all_good
+
+    def write_results(self):
+        """Create results.txt file, and write to it if write_func is set.
+        The file signifies that the simulation ran to completion."""
+        if mwxrun.me == 0:
+            mwxutil.mkdir_p(WarpXDiagnostic.DIAG_DIR)
+            with open(os.path.join(WarpXDiagnostic.DIAG_DIR, "results.txt"), 'a') as results_file:
+                if callable(self._write_func):
+                    sys.stdout = results_file
+                    self._write_func()
```

Comment:
I think it is important that this not be called only on the root since conceivably we might want to track a quantity that requires summing over processors. My take it that instead of allowing the passed function to directly write to `stdout` it should return something (a string, a float an int, whatever) and that gets written to `results.txt` by the root. Does that make sense?